topsf/config/run3/config_sf.py

```diff
@@ -0,0 +1,1414 @@
+# coding: utf-8
+
+"""
+++++++++++ WIP ++++++++++
+Configuration creation for top-tagging scale factor
+derivation using Run3 samples.
+++++++++++ WIP ++++++++++
+"""
+
+from __future__ import annotations
+
+import functools
+import os
+
+import order as od
+import yaml
+
+from scinum import Number
+
+from columnflow.util import DotDict
+from columnflow.config_util import (
+    add_shift_aliases,
+    get_root_processes_from_campaign,
+    get_shifts_from_sources,
+    verify_config_processes,
+)
+
+from topsf.config.variables import add_variables
+from topsf.config.categories import add_categories
+
+thisdir = os.path.dirname(os.path.abspath(__file__))
+
+
+def add_config(
+    analysis: od.Analysis,
+    campaign: od.Campaign,
+    config_name: str | None = None,
+    config_id: int | None = None,
+    limit_dataset_files: int | None = None,
+) -> od.Config:
+    """
+    Configurable function for creating a config for a run3 analysis given
+    a base *analysis* object and a *campaign* (i.e. set of datasets).
+    """
+    # validation
+    assert campaign.x.year in [2022, 2023]
+    if campaign.x.year == 2022:
+        assert campaign.x.EE in ["pre", "post"]
+    elif campaign.x.year == 2023:
+        assert campaign.x.BPix in ["pre", "post"]
+
+    # gather campaign data
+    year = campaign.x.year
+    year2 = year % 100
+    corr_postfix = ""
+    if year == 2022:
+        corr_postfix = f"{campaign.x.EE}EE"
+    elif year == 2023:
+        corr_postfix = f"{campaign.x.BPix}BPix"
+
+    implemented_years = [2022]
+
+    if year not in implemented_years:
+        raise NotImplementedError("For now, only 2022 campaign is fully implemented")
+
+    # create a config by passing the campaign
+    # (if id and name are not set they will be taken from the campaign)
+    cfg = analysis.add_config(campaign, name=config_name, id=config_id)
+
+    # add some important tags to the config
+    cfg.x.cpn_tag = f"{year}{corr_postfix}"
+
+    if year in (2022, 2023):
+        cfg.x.run = 3
```

Comment:
I think the `run` aux can be set directly here since this is by definition a Run3 config.
```suggestion
    cfg.x.run = 3
```